lib/SILGen/SILGenType.cpp

```diff
@@ -781,6 +793,10 @@ SILFunction *SILGenModule::emitProtocolWitness(
                  derivativeId->getParameterIndices()->getString();
   }
 
+  if (witnessRef.isDistributedThunk()) {
+    nameBuffer = nameBuffer + "TE"; // TE signifies a distributed thunk.
```

Comment:
This should be handled in `mangleWitnessThunk`.

Reply:
Okey!

Reply:
So this one turns out is not trivial -- AutoDiff also has the issue here: the ASTMangler takes only AST types and we'd need to give it a SILDeclRef... The AutoDiff mangling does it ad hoc here for the same reason I guess. 

It feels like a layering problem to pass SIL stuff to an AST type...? But perhaps it's okey and I should add an optional SILDeclRef there or something similar?